Framework/Algorithms/src/XrayAbsorptionCorrection.cpp

```diff
@@ -0,0 +1,234 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidAlgorithms/XrayAbsorptionCorrection.h"
+#include "MantidAPI/Axis.h"
+#include "MantidAPI/Sample.h"
+#include "MantidAPI/SpectrumInfo.h"
+#include "MantidAPI/WorkspaceUnitValidator.h"
+#include "MantidKernel/BoundedValidator.h"
+#include "MantidKernel/CompositeValidator.h"
+#include "MantidKernel/Material.h"
+#include "MantidKernel/PhysicalConstants.h"
+#include "MantidKernel/VectorHelper.h"
+#include <iostream>
+#include <math.h>
+#include <numeric>
+
+using namespace Mantid::API;
+using namespace Mantid::Geometry;
+using namespace Mantid::Kernel;
+using Mantid::DataObjects::Workspace2D;
+namespace PhysicalConstants = Mantid::PhysicalConstants;
+
+/// @cond
+namespace {
+
+constexpr double DEFAULT_ANGLE = 45.0;
+constexpr double DEFAULT_DETECTOR_DISTANCE = 10.0;
+
+} // namespace
+/// @endcond
+
+namespace Mantid {
+namespace Algorithms {
+
+DECLARE_ALGORITHM(XrayAbsorptionCorrection)
+
+/**
+ * Initialize the algorithm
+ */
+void XrayAbsorptionCorrection::init() {
+
+  auto wsValidator = std::make_shared<CompositeValidator>();
+
+  declareProperty(std::make_unique<WorkspaceProperty<>>(
+                      "InputWorkspace", "", Direction::Input, wsValidator),
+                  "The name of the input workspace.");
+
+  declareProperty(
+      std::make_unique<WorkspaceProperty<>>("MuonImplantationProfile", "",
+                                            Direction::Input, wsValidator),
+      "The name of the Muon Implantation Profile.");
+
+  declareProperty(std::make_unique<WorkspaceProperty<>>("OutputWorkspace", "",
+                                                        Direction::Output),
+                  "The name to use for the output workspace.");
+
+  auto positiveDouble = std::make_shared<Kernel::BoundedValidator<double>>();
+  positiveDouble->setLower(0);
+  declareProperty("DetectorAngle", DEFAULT_ANGLE, positiveDouble,
```

Comment:
In the other absorption correction algorithms, the detector distance\angle is read from the instrument definition rather than being set up as input parameters. The other algorithms calculate the corrections for all detectors by default - although you can specify a subset of detectors if you prefer. Have you been specifically asked to set it up like this where you take the position info in as parameters?

Reply:
I have been asked to set it up like this

Reply:
the input files are just aski at the moment and there is no IDF